File: simpleweb/simpleweb.hs
{-# LANGUAGE OverloadedStrings #-}

module Main where

import Network.Wai
import Network.Wai.Handler.Warp
import Network.HTTP.Types

app :: Application
app req sendResp = sendResp $
  case pathInfo req of
    ["hello"] -> responseBuilder
      status200
      [("Content-Type", "text/plain")]
      "Hello World\n"

    _ -> responseBuilder
      status404
      [("Content-Type", "text/plain")]
      "404 NOT FOUND\n"

main :: IO ()
main = run 8080 app

File: simpleweb/simpleweb.go
package main

import (
	"fmt"
	"net/http"
	"os"
)

func app(w http.ResponseWriter, r *http.Request) {
	switch r.URL.EscapedPath() {
	case "/hello":
		w.Header().Set("Content-Type", "text/plain")
		w.WriteHeader(200)
		fmt.Fprintln(w, "Hello World")

	default:
		w.Header().Set("Content-Type", "text/plain")
		w.WriteHeader(404)
		fmt.Fprintln(w, "404 NOT FOUND")
	}
}

func main() {
	if err := http.ListenAndServe(":8080", http.HandlerFunc(app)); err != nil {
		fmt.Fprintln(os.Stderr, err.Error())
		os.Exit(1)
	}
}
